src/storage/src/statistics.rs

```diff
@@ -71,7 +71,8 @@ impl SourceStatisticsMetricsDefinitions {
                 help: "The number of bytes of the source envelope state kept. This will be specific to the envelope in use.",
                 var_labels: ["source_id", "worker_id", "parent_source_id", "shard_id"],
             )),
-            envelope_state_count: registry.register(metric!(
+            envelope_state_records: registry.register(metric!(
+                // The different name for the metric itself is historical.
                 name: "mz_source_envelope_state_count",
```

Comment:
In Compute, I'd prefer to change the metric name, too. It's relatively simple to work-around the name change in Grafana, we just need to be aware where it's used from and duplicate the query with the new name. But up to you!

Reply:
Ack, I've gone ahead and done that. @moulimukherjee / @guswynn: let me know if you'd prefer the original approach of leaving the metric name unchanged.

Reply:
I prefer it all to match!

Reply:
Great!